modules/apps/commerce/commerce-shipping-engine-fixed-service/src/main/java/com/liferay/commerce/shipping/engine/fixed/internal/search/CommerceShippingFixedOptionIndexer.java

```diff
@@ -0,0 +1,223 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.shipping.engine.fixed.internal.search;
+
+import com.liferay.commerce.shipping.engine.fixed.model.CommerceShippingFixedOption;
+import com.liferay.commerce.shipping.engine.fixed.service.CommerceShippingFixedOptionLocalService;
+import com.liferay.portal.kernel.dao.orm.IndexableActionableDynamicQuery;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.search.BaseIndexer;
+import com.liferay.portal.kernel.search.BooleanClauseOccur;
+import com.liferay.portal.kernel.search.BooleanQuery;
+import com.liferay.portal.kernel.search.Document;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.IndexWriterHelper;
+import com.liferay.portal.kernel.search.Indexer;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.search.Summary;
+import com.liferay.portal.kernel.search.filter.BooleanFilter;
+import com.liferay.portal.kernel.search.filter.TermFilter;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+import java.util.Locale;
+
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alec Sloan
+ */
+@Component(enabled = false, immediate = true, service = Indexer.class)
+public class CommerceShippingFixedOptionIndexer
+	extends BaseIndexer<CommerceShippingFixedOption> {
+
+	public static final String CLASS_NAME =
+		CommerceShippingFixedOption.class.getName();
+
+	public CommerceShippingFixedOptionIndexer() {
+		setFilterSearch(true);
+	}
+
+	@Override
+	public String getClassName() {
+		return CLASS_NAME;
+	}
+
+	@Override
+	public void postProcessContextBooleanFilter(
+			BooleanFilter contextBooleanFilter, SearchContext searchContext)
+		throws Exception {
+
+		long commerceShippingMethodId = (long)searchContext.getAttribute(
+			"commerceShippingMethodId");
+
+		if (commerceShippingMethodId != -1) {
+			TermFilter termFilter = new TermFilter(
+				"commerceShippingMethodId",
+				String.valueOf(commerceShippingMethodId));
+
+			contextBooleanFilter.add(termFilter, BooleanClauseOccur.MUST);
+		}
+	}
+
+	@Override
+	public void postProcessSearchQuery(
+			BooleanQuery searchQuery, BooleanFilter fullQueryBooleanFilter,
+			SearchContext searchContext)
+		throws Exception {
+
+		super.postProcessSearchQuery(
+			searchQuery, fullQueryBooleanFilter, searchContext);
+
+		addSearchTerm(searchQuery, searchContext, Field.ENTRY_CLASS_PK, false);
+		addSearchTerm(
+			searchQuery, searchContext, "commerceShippingMethodId", false);
+		addSearchLocalizedTerm(searchQuery, searchContext, Field.NAME, true);
```

Comment:
Sort constant chunks before string.

```suggestion
		addSearchLocalizedTerm(searchQuery, searchContext, Field.NAME, true);
		addSearchTerm(
			searchQuery, searchContext, "commerceShippingMethodId", false);
```